### ISSUE TITLE
Implement this into SwiftUI

### ISSUE THREAD

**Issue by reinaldoriant** [danielsaidi/Sheeeeeeeeet#151]
Do you have sample to use this in SwiftUI?

**Comment by danielsaidi** [comment]
Hi! No, I haven’t tried this yet, but I think it would be pretty easy to achieve.

**Comment by bukira** [comment]
Any help on this?

**Comment by danielsaidi** [comment]
I had a look at this today, and I'm not really sure how to extend this to SwiftUI. I mean, the custom action sheet would be great to have, but the context menu and alert parts are already easily available in SwiftUI.

I guess creating a SwiftUI bridge for the custom action sheet would be good, but I'm not sure when I'll have time to do it.

**Comment by bukira** [comment]
yeah tis the custom action sheet that I use, this for looking I have found a few other kind of similar things so will try hacking them to suit my uses

**Comment by bukira** [comment]
for the custom action sheet, not sure if this is what you were thinking off but I did the following


struct CustomActionSheet: UIViewControllerRepresentable {
    func makeUIViewController(context: Context) -> CustomActionSheetController {
        let customActionSheet = CustomActionSheetController()
        customActionSheet.delegate = context.coordinator
        customActionSheet.sheet = self.sheet
        return customActionSheet
    }

    func updateUIViewController(_ uiViewController: CustomActionSheetController, context: Context) {}

    typealias UIViewControllerType = CustomActionSheetController
    @Binding var isShowing: Bool
    var sheet: Sheeeeeeeeet.ActionSheet?

    class Coordinator: NSObject, CustomActionSheetControllerDelegate {
        func actionSheetDidFinish(_ customActionSheet: CustomActionSheetController) {
            parent.isShowing = false
        }
        var parent: CustomActionSheet
        init(_ parent: CustomActionSheet) {
            self.parent = parent
        }
    }

    func makeCoordinator() -> Coordinator {
        Coordinator(self)
    }
}

protocol CustomActionSheetControllerDelegate: AnyObject {
    func actionSheetDidFinish(_ customActionSheet: CustomActionSheetController)
}

class CustomActionSheetController: UIViewController {
    weak var delegate: CustomActionSheetControllerDelegate?
    var sheet: Sheeeeeeeeet.ActionSheet?
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        sheet?.present(in: self, from: self.view) {
            self.sheet?.presenter?.events.didDismissWithBackgroundTap = { self.delegate?.actionSheetDidFinish(self) }
        }
    }
}

**Comment by danielsaidi** [comment]
@bukira Sorry for the slow reply. Did that work in your case?

**Comment by bukira** [comment]
yes this did the job for me, feel free to add to the repo if this is what you had in mind also

**Comment by danielsaidi** [comment]
Wow, perfect! I will try to get around to it, but I have so many other things going on right now, that it may take a while.

**Comment by bukira** [comment]
np, the above works for me fine and anyone else who wants to use it, works seamlessly and perfectly, a win win

**Comment by danielsaidi** [comment]
That's great, thank you. I'll keep this issue opened until the code is in the master branch.
